rnd/autogpt_server/autogpt_server/server/routers/analytics.py

```diff
@@ -0,0 +1,138 @@
+"""Analytics API"""
+
+from typing import Annotated, Optional
+
+import fastapi
+import prisma
+import prisma.enums
+import pydantic
+
+import autogpt_server.data.analytics
+from autogpt_server.server.utils import get_user_id
+
+router = fastapi.APIRouter()
+
+
+class UserData(pydantic.BaseModel):
+    user_id: str
+    email: str
+    name: str
+    username: str
+
+
+@router.post(path="/log_new_user")
+async def log_create_user(
+    user_id: Annotated[str, fastapi.Depends(get_user_id)],
+    user_data: Annotated[
+        UserData, fastapi.Body(..., embed=True, description="The user data to log")
+    ],
+):
+    """
+    Log the user ID for analytics purposes.
+    """
+
+    result = await autogpt_server.data.analytics.log_raw_analytics(
+        user_id,
+        prisma.enums.AnalyticsType.CREATE_USER,
+        user_data.model_dump(),
+        "",
+    )
+    return result.id
+
+
+@router.post(path="/log_tutorial_step")
+async def log_tutorial_step(
+    user_id: Annotated[str, fastapi.Depends(get_user_id)],
```

Comment:
Why are we not using google analytics events functionality. We can then programatically generate reports, without needing to reinvent frontend analytics.

Reply:
We have a frontend that has a backend. I don't know who thought introducing Google Analytics was sufficient but it was incorrect.

Reply:
but we don't use the backend as such with nextjs right, so how come GA wouldn't be enough in this case?

Reply:
If a server action has an error GA won’t track it

Reply:
@aarushik93  I got this error: 
<img width="963" alt="image" src="https://github.com/user-attachments/assets/b1ffbe0e-e506-4e18-8ab0-8a27a555ecc6">

Does that mean most of the components were server components unless it's explicitly mentioned?

Reply:
Yes, by default everything is a server component.